skiko/src/awtMain/cpp/linux/swingRedrawer.cc

```diff
@@ -0,0 +1,187 @@
+#include <jawt_md.h>
+#include <GL/gl.h>
+#include <GL/glx.h>
+#include <X11/X.h>
+#include <X11/Xlib.h>
+#include <X11/Xresource.h>
+#include <cstdlib>
+#include <unistd.h>
+#include <stdio.h>
+#include "jni_helpers.h"
+
+class OffScreenContext
+{
+public:
+    Display* display;
+    GLXContext context;
+    GLXFBConfig* fbConfigs;
+
+    OffScreenContext() {
+        const int glxContextAttribs[] {
+            GLX_DRAWABLE_TYPE, GLX_WINDOW_BIT,
+            GLX_RENDER_TYPE, GLX_RGBA_BIT,
+            GLX_DOUBLEBUFFER, False,
+            GLX_RED_SIZE, 8,
+            GLX_GREEN_SIZE, 8,
+            GLX_BLUE_SIZE, 8,
+            None
+        };
+
+        display = XOpenDisplay(nullptr);
+
+        int numConfigs = 0;
+        fbConfigs = glXChooseFBConfig(display, DefaultScreen(display), glxContextAttribs, &numConfigs);
+
+        XVisualInfo* visual = glXGetVisualFromFBConfig(display, fbConfigs[0]);
+
+        context = glXCreateContext(display, visual, nullptr, True);
+
+        XFree(visual);
+    }
+
+    ~OffScreenContext() {
+        XFree(fbConfigs);
+        glXDestroyContext(display, context);
+        XCloseDisplay(display);
+    }
+};
+
+class OffScreenBuffer
+{
+public:
+    Display* display;
+    GLXPbuffer pbuffer;
+    int width;
+    int height;
+
+    OffScreenBuffer(OffScreenContext* context, int _width, int _height) {
+        int pbufferAttribs[] = {
+            GLX_PBUFFER_WIDTH, width,
+            GLX_PBUFFER_HEIGHT, height,
+            None  
+        };
+
+        width = _width;
+        height = _height;
+        display = context->display;
+
+        pbuffer = glXCreatePbuffer(context->display, context->fbConfigs[0], pbufferAttribs);
```

Comment:
https://registry.khronos.org/OpenGL-Refpages/gl2.1/xhtml/glXCreatePbuffer.xml can generate error. Should we process that?